File: HFSGuardaDiretorio_SharpDevelop_Ruby/comum/ArquivoLog.rb
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.IO, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Text, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module HFSGuardaDiretorio.comum
	class ArquivoLog < Log
		def FileName
			return self.@_FileName
		end

		def FileName=(value)
			self.@_FileName = value
		end

		def FileLocation
			return self.@_FileLocation
		end

		def FileLocation=(value)
			self.@_FileLocation = value
			if self.@_FileLocation.LastIndexOf("\\") != (self.@_FileLocation.Length - 1) then
				self.@_FileLocation
			end
		end

		def initialize()
			@_FileName = ""
			@_FileLocation = ""
			self.FileLocation = "C:\\"
			self.FileName = "mylog.txt"
		end

		def RecordMessage(Message, Severity)
			self.RecordMessage(Message.Message, Severity)
		end

		def RecordMessage(Message, Severity)
			fileStream = nil
			writer = nil
			message = StringBuilder.new()
			begin
				fileStream = FileStream.new(self.@_FileLocation + self.@_FileName, FileMode.OpenOrCreate, FileAccess.Write)
				writer = StreamWriter.new(fileStream)
				writer.BaseStream.Seek(0, SeekOrigin.End)
				message.Append(System.DateTime.Now.ToString()).Append(",").Append(Message)
				writer.WriteLine(message.ToString())
				writer.Flush()
			ensure
				if writer != nil then
					writer.Close()
				end
			end
		end
	end
end

File: HFSGuardaDiretorio_SharpDevelop_Python/comum/ArquivoLog.py
from System import *
from System.IO import *
from System.Text import *

class ArquivoLog(Log):
	def get_FileName(self):
		return self.__FileName

	def set_FileName(self, value):
		self.__FileName = value

	FileName = property(fget=get_FileName, fset=set_FileName)

	def get_FileLocation(self):
		return self.__FileLocation

	def set_FileLocation(self, value):
		self.__FileLocation = value
		if self.__FileLocation.LastIndexOf("\\") != (self.__FileLocation.Length - 1):
			self.__FileLocation += "\\";

	FileLocation = property(fget=get_FileLocation, fset=set_FileLocation)

	def __init__(self):
		self.__FileName = ""
		self.__FileLocation = ""
		self.FileLocation = "C:\\"
		self.FileName = "mylog.txt"

	def RecordMessage(self, Message, Severity):
		self.RecordMessage(Message.Message, Severity)

	def RecordMessage(self, Message, Severity):
		fileStream = None
		writer = None
		message = StringBuilder()
		try:
			fileStream = FileStream(self.__FileLocation + self.__FileName, FileMode.OpenOrCreate, FileAccess.Write)
			writer = StreamWriter(fileStream)
			writer.BaseStream.Seek(0, SeekOrigin.End)
			message.Append(System.DateTime.Now.ToString()).Append(",").Append(Message)
			writer.WriteLine(message.ToString())
			writer.Flush()
		finally:
			if writer != None:
				writer.Close()